src/main/java/org/opentripplanner/middleware/connecteddataplatform/AnonymizedTripRequest.java

```diff
@@ -137,55 +134,98 @@ private List<AnonymizedItinerary> getItineraries(
+                    anonymizedLeg.duration = leg.duration;
+                    anonymizedLeg.startTime = leg.startTime;
+                    anonymizedLeg.endTime = leg.endTime;
+                    anonymizedLeg.mode = leg.mode;
+                    anonymizedLeg.transitLeg = leg.transitLeg;
+                    anonymizedLeg.fromStop = leg.from.stopId;
+                    anonymizedLeg.from = new Coordinates(leg.from.lat, leg.from.lon);
+                    anonymizedLeg.toStop = leg.to.stopId;
+                    anonymizedLeg.to = new Coordinates(leg.to.lat, leg.to.lon);
+                    if (Boolean.TRUE.equals(leg.transitLeg)) {
+                        // Parameters for a transit leg.
+                        anonymizedLeg.agencyId = leg.agencyId;
+                        anonymizedLeg.interlineWithPreviousLeg = leg.interlineWithPreviousLeg;
+                        anonymizedLeg.realTime = leg.realTime;
+                        anonymizedLeg.routeId = leg.routeId;
+                        anonymizedLeg.routeShortName = leg.routeShortName;
+                        anonymizedLeg.routeLongName = leg.routeLongName;
+                        anonymizedLeg.routeType = leg.routeType;
+                        anonymizedLeg.tripBlockId = leg.tripBlockId;
+                        anonymizedLeg.tripId = leg.tripId;
+                    } else {
+                        // Parameters for non transit leg.
+                        anonymizedLeg.rentedVehicle = leg.rentedVehicle;
+                    }
+                    itin.legs.add(anonymizedLeg);
                 }
-                itin.legs.add(anonymizedLeg);
             }
             anonymizedItineraries.add(itin);
         }
         return anonymizedItineraries;
     }
 
     /**
-     * Create {@link Coordinates} for a leg. Replace lat/lon values with the lat/lon values created for the trip
-     * request if not a transit leg. The start and end legs will then be consistent with the trip's 'to' and 'from'
-     * place.
+     * Define the coordinates for all legs. If the leg is a walking leg and is before the first transit leg or after
+     * the last transit leg the coordinates must not be provided for privacy. Any transit leg or walking legs between
+     * transits legs, the coordinates can be provided.
      */
-    private Coordinates getLegCoordinates(
-        boolean isTransitLeg,
-        boolean isFirstOrLastLegOfTrip,
-        Place place,
-        Coordinates tripRequestCoordinates
-    ) {
-        return (!isTransitLeg && isFirstOrLastLegOfTrip) ? tripRequestCoordinates : new Coordinates(place.lat, place.lon);
+    private void processLegCoordinates(List<Leg> legs) {
+        int firstTransitLeg = getFirstTransitLeg(legs);
+        int lastTransitLeg = getLastTransitLeg(legs);
+        for (int i = 0; i <= legs.size() -1; i++) {
+            Leg leg = legs.get(i);
+            if (
+                leg.mode.equalsIgnoreCase("walk") &&
```

Comment:
All non-transit legs should have their locations stripped! Is there any way to use the `isTransit` leg property?

Reply:
Yep! Updated.